File: Week 2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

float formula(int words, int letters, int punctuation);

int main(void)
{
    string input = get_string("Input text: ");

    // Get amount of characters per input (including \0)
    int n = strlen(input);
    
    // Introduction of variables
    int words = 0;
    int letters = 0;
    int punctuation = 0;

    // Check every character
    for (int i = 0; i < n; i++)
    {
        // If it is a letter
        if (isalpha(input[i]) != 0)
        {
            letters ++;
        }
        else
        {
            // If it is a space or the last letter
            if (input[i] == ' ' || input[i + 1] == '\0')
            {
                words ++;
            }    
            
            // If it is the end of a sentence
            if (input[i] == '.' || input[i] == '?' || input[i] == '!')
            {
                punctuation ++;
            }
        }
    }

    // Get index from function
    float index = formula(words, letters, punctuation);
    
    // Round value
    int grade = round(index);
    
    // Print result
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}

float formula(int words, int letters, int punctuation)
{
    float L = letters / (float) words * 100.0;
    float S = punctuation / (float) words * 100.0;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    
    return index;
}

File: Week 6/pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

words = 0
sentences = 0
letters = 0

# check every character
for i in text:
    if i.isalpha() == True:
        letters += 1
    elif i == " ":
        words += 1
    elif i == "." or i == "!" or i == "?":
        sentences += 1
if words != 0:
    words += 1

# formula to calculate the reading level
L = letters / words * 100
S = sentences / words * 100
formula = 0.0588 * L - 0.296 * S - 15.8

# round level to next integer
grade = int(round(formula))

# decision
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")